### PR TITLE
change all ID parameters in URI to UUID.

### LINKED ISSUE #2204: change all ID parameters in URI to UUID
Currently we use string for the ID field in some places and UUID field in someplaces. When we are using string, we are again internally creating UUID out of it.  This can avoided. 

- [ ] Check all controller classes and set ID to type UUID
- [ ] fix the audit annotation to pass the correct type.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes #2204

### CODE REVIEW

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by profoundwu** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java [hunk 1]
```diff
@@ -146,9 +145,9 @@ public Csp getCspFromServicePolicyId(String id) {
      * @param taskId id of migrate task.
      * @return csp.
      */
-    public Csp getCsFromWorkflowTaskId(String taskId) {
+    public Csp getCspFromWorkflowTaskId(UUID taskId) {
         try {
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java:149]: The taskId in workflow is not a UUID. Just keep type String

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java [hunk 2]
```diff
@@ -67,31 +68,33 @@ public List<WorkFlowTask> queryTasks(
...
     @Tag(name = "Workflow", description = "APIs to manage the Workflow")
     @Operation(description = "Complete tasks by task ID and set global process variables .")
-    @PutMapping(value = "/workflow/complete/task/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @PutMapping(
+            value = "/workflow/complete/task/{taskId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
-    @AuditApiRequest(methodName = "getCsFromWorkflowTaskId")
+    @AuditApiRequest(methodName = "getCsFromWorkflowTaskId", paramTypes = UUID.class)
     public void completeTask(
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java:76]: The taskId in workflow is not a UUID. Just keep type String

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java (hunk 1 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/config/GetCspInfoFromRequest.java:149]: updated

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java (hunk 2 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java:75]: updated.

**Comment by luoping-12345** [comment]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**